woocommerce/woocommerce
# Implement the new color picker input and preview for integrations

As mentioned on the previous color picker PR https://github.com/woothemes/woocommerce/pull/8356#issuecomment-111146310 - integrations uses different code to generate their settings. It probably makes sense to put some effort towards using one unified settings API, but that is a bigger undertaking. In the mean time, this PR updates `generate_color_html` to use the new picker.

before:

![before integration color picker](https://cldup.com/w58WVrWceD-3000x3000.png)

after:

![after integration color picker](https://cldup.com/XYJ92jwRZW-3000x3000.png)

props @shivapoudel 

**Testing Steps:**
- Activate an integration/plugin that uses color settings, or install the [woocommerce-google-analytics-integration](https://github.com/woothemes/woocommerce-google-analytics-integration) plugin and modify its `init_form_fields` method to include a color option:

``` php
            'ga_color' => array(
                'title'    => __( 'Base Colour', 'woocommerce' ),
                'desc'     => __( 'The base colour for WooCommerce email templates. Default <code>#557da1</code>.', 'woocommerce' ),
                'id'       => 'woocommerce_email_base_color',
                'type'     => 'color',
                'css'      => 'width:6em;',
                'default'  => '#557da1',
                'autoload' => false
            ),
```
- View integration settings and verify that the new color picker is used.
